### PR TITLE
Clang-format: increase column limit from 90 to 105

### LINKED ISSUE #10740: [Code Style] Increase line length limit
The Minetest coding style has a soft line length limit at 80 characters and a hard line length limit at 90 characters. I really hate this.
Guys, we're not in the 1970's anymore. I think it's safe to raise the line length limit to … i don't know, 140 maybe? (feel free to suggest another number)

So many PRs get delayed because some lines are slightly too long. So annoying. -_-
I also think this low line length limit sometimes makes code readability *worse* because it sometimes forces too many linebreaks.

Anyway, note this issue is only a policy suggestion. I do not suggest to touch existing code, but only to reconsider the policy for *future code*.

### PR DESCRIPTION
**Pull request by Zughy**
Closes #10740 

* The majority of core devs agree, celeron included
* As the previous one was 90, i.e. 10 more than the soft limit (80), and the soft limit core devs agreed one is now 95, I applied the same difference (95 + 10 = 105)

## To do

This PR is Ready for Review.

### CODE REVIEW

**Comment by Desour** [comment]
While you are at it, would you mind fixing the indent width?

https://dev.minetest.net/Code_style_guidelines#Don.27t_let_things_get_too_large
>(Note that this column count assumes 4-space indents.)

**Comment by Zughy** [comment]
@Desour I'm honestly super confused, I realise this is bigger than me https://irc.minetest.net/minetest-dev/2021-01-11#i_5772268

**Comment by sfan5** [comment]
Change the config to say `IndentWidth: 4`

File: .clang-format [hunk 1]
```diff
@@ -17,7 +17,7 @@ FixNamespaceComments: false
 AllowShortIfStatementsOnASingleLine: false
 IndentCaseLabels: false
 AccessModifierOffset: -8
```
Comment (rubenwardy) [inline .clang-format:19]: -4

Comment (rubenwardy) [reply .clang-format:19]: also 

ContinuationIndentWidth: 8
ConstructorInitializerIndentWidth: 8

Comment (Zughy) [reply .clang-format]: Done

**Comment by sfan5** [comment]
Now clang-format wants to realign everything that was put on multiple lines due to the column limit before.

What a shitty mess...

**Comment by rubenwardy** [comment]
Great, clang-format runs differently locally to on CI

File: src/script/lua_api/l_modchannels.cpp [hunk 2]
```diff
@@ -108,7 +107,7 @@ void ModChannelRef::Register(lua_State *L)
 	lua_pop(L, 1); // Drop metatable
 
 	luaL_openlib(L, 0, methods, 0); // fill methodtable
-	lua_pop(L, 1);			// Drop methodtable
+	lua_pop(L, 1);					// Drop methodtable
```
Comment (sfan5) [inline src/script/lua_api/l_modchannels.cpp:110]: ?????????????

File: src/util/sha2.h [hunk 3]
```diff
@@ -100,10 +100,10 @@ extern "C" {
 #endif
 
 #define SHA_LBLOCK 16
-#define SHA_CBLOCK                                                                       \
-	(SHA_LBLOCK * 4) /* SHA treats input data as a                                   \
-			  * contiguous array of 32 bit wide                              \
-			  * big-endian values. */
+#define SHA_CBLOCK                                                                                      \
+	(SHA_LBLOCK * 4) /* SHA treats input data as a                                                      \
+					  * contiguous array of 32 bit wide                                                 \
```
Comment (sfan5) [inline src/util/sha2.h:105]: wtf is this

File: src/content/mods.cpp [hunk 4]
```diff
@@ -455,9 +437,9 @@ bool ModMetadata::load(const std::string &root_path)
 
 	if (!Json::parseFromStream(builder, is, &root, &errs)) {
 		errorstream << "ModMetadata[" << m_mod_name
-			    << "]: failed read data "
-			       "(Json decoding failure). Message: "
-			    << errs << std::endl;
+					<< "]: failed read data "
```
Comment (sfan5) [inline src/content/mods.cpp:440]: this should be two (or one?) but now it's three

**Review by sfan5 — CHANGES_REQUESTED** [review]

**Comment by sfan5** [comment]
:-1:
Letting clang-format make hundred lines of shit changes to our codebase (some of which aren't even correct) for **no** reason is not acceptable.
Badly working linting is clearly worse than no linting at all so I suggest finally removing the cancer that is clang-format from our CI.

**Comment by rubenwardy** [comment]
clang-format is broken on my computer currently, I suggest that someone else tries to run it

**Comment by sfan5** [comment]
If in the years we've had it nobody has been able to configure it to exactly match our codestyle I can only assume it is impossible.

**Comment by rubenwardy** [comment]
We need to change our codestyle to be less arbitrary. For example: we use two tabs of indentation for continuation, except for statements where one is used

**Comment by rubenwardy** [comment]
I think clang-format as it is is damaging and should be removed if it stays this way. But automatic formatting is a massive productivity booster because it removes the need to argue over style

**Comment by numberZero** [comment]
> But automatic formatting is a massive productivity booster because it removes the need to argue over style

Then what e.g. #10740 was all about?

**Comment by rubenwardy** [comment]
> Then what e.g. #10740 was all about?

That's not a PR. I meant in terms of code review

**Comment by Zughy** [comment]
Not the right time, as MT clang right now is horrible. Closing

**Comment by numberZero** [comment]
@Zughy What’s about doing exactly what the title says instead of all that stuff in this PR?

**Comment by sfan5** [comment]
There's three options here, only one of which is acceptable:
* Increase the clang-format column limit, don't adjust the code and live with failing CI forever
* Increase the clang-format column limit, adjust all the code and cause hundreds of unnecessary changes
* Do not touch the clang-format config (and instead consider #10863)

If it turns out that all these formatting changes are not due to `ColumnLimit: 105` but the other config changes I will gladly approve a PR for that.

**Pull request closed without merging**